tests/jvm/src/test/scala/cats/effect/IOPlatformSpecification.scala

```diff
@@ -531,6 +532,25 @@ trait IOPlatformSpecification { self: BaseSpec with ScalaCheck =>
           runtime.shutdown()
         }
       }
+
+      if (javaMajorVersion >= 21)
+        "block in-place on virtual threads" in real {
+          val loomExec = classOf[Executors]
+            .getDeclaredMethod("newVirtualThreadPerTaskExecutor")
+            .invoke(null)
+            .asInstanceOf[ExecutorService]
+
+          val loomEc = ExecutionContext.fromExecutor(loomExec)
+
+          IO.blocking {
+            classOf[Thread]
+              .getDeclaredMethod("isVirtual")
+              .invoke(Thread.currentThread())
+              .asInstanceOf[Boolean]
```

Comment:
```suggestion
            Thread.currentThread().asInstanceOf[{ def isVirtual: Boolean }].isVirtual
```

Is a bit more concise :)